test/charms/main/lib/charm.py

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env python3
+
+import os
+import base64
+import pickle
+
+from juju.framework import (
+    EventBase,
+    Event,
+)
+
+from juju.charm import (
+    CharmBase,
+    CharmEvents,
+)
+
+import logging
+
+logger = logging.getLogger()
+
+
+class DBRelationJoinedEvent(EventBase):
+    pass
+
+
+class DBRelationChangedEvent(EventBase):
+    pass
+
+
+class DBRelationDepartedEvent(EventBase):
+    pass
+
+
+class DBRelationBrokenEvent(EventBase):
+    pass
+
+
+class MonRelationJoinedEvent(EventBase):
+    pass
+
+
+class MonRelationChangedEvent(EventBase):
+    pass
+
+
+class MonRelationDepartedEvent(EventBase):
+    pass
+
+
+class MonRelationBrokenEvent(EventBase):
+    pass
+
+
+class HARelationJoinedEvent(EventBase):
+    pass
+
+
+class HARelationChangedEvent(EventBase):
+    pass
+
+
+class HARelationDepartedEvent(EventBase):
+    pass
+
+
+class HARelationBrokenEvent(EventBase):
+    pass
+
+
+class MainCharmEvents(CharmEvents):
+    db_relation_joined = Event(DBRelationJoinedEvent)
+    db_relation_changed = Event(DBRelationChangedEvent)
+    db_relation_departed = Event(DBRelationDepartedEvent)
+    db_relation_broken = Event(DBRelationBrokenEvent)
+    mon_relation_joined = Event(MonRelationJoinedEvent)
+    mon_relation_changed = Event(MonRelationChangedEvent)
+    mon_relation_departed = Event(MonRelationDepartedEvent)
+    mon_relation_broken = Event(MonRelationBrokenEvent)
+    ha_relation_joined = Event(HARelationJoinedEvent)
+    ha_relation_changed = Event(HARelationChangedEvent)
+    ha_relation_departed = Event(HARelationDepartedEvent)
+    ha_relation_broken = Event(HARelationBrokenEvent)
```

Comment:
Looks good for now.. We'll want to use something more dynamic later.

Reply:
Removed it as Cory's PR was merged.